### PR TITLE
Client side failover

### PR DESCRIPTION
**Pull request by shelocks**
#182

### CODE REVIEW

**Comment by trustin** [comment]
@shelocks Thanks for another contribution! Let me review soon.

**Comment by trustin** [comment]
Could we let user specify backoff strategy? Quick googling gives me the following types, which seems good and generic enough:

- [BackoffStrategy](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/retry/RetryPolicy.BackoffStrategy.html) instead of retrying immediately
- [RetryConfition](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/retry/RetryPolicy.RetryCondition.html) instead of using integer parameter for max try count and hard-coded exception filter

What do you think?

**Comment by trustin** [comment]
@anuraaga PTAL as well. We could also add a few more decorating cients to achieve what you want?

File: src/main/java/com/linecorp/armeria/client/routing/FailOverClient.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.DecoratingClient;
+
+import java.util.function.Function;
+
+public final class FailOverClient extends DecoratingClient {
```
Comment (anuraaga) [inline src/main/java/com/linecorp/armeria/client/routing/FailOverClient.java]: RetryingClient instead of FailOver?

Comment (trustin) [inline src/main/java/com/linecorp/armeria/client/routing/FailOverClient.java]: +1

**Comment by shelocks** [comment]
@trustin Sorry for the late reply.For RetryPolicy,LGTM,I will update code later.Thanks!

**Comment by trustin** [comment]
@shelocks Thanks for coming back! Looking forward the update.

**Comment by trustin** [comment]
@shelocks Gentle ping? :-)

**Comment by trustin** [comment]
Let me close this PR for now. Thanks for trying to contribute!

**Pull request closed without merging**